packages/react/src/components/Popup/createPopperReferenceProxy.ts

```diff
@@ -0,0 +1,37 @@
+import { isRefObject, toRefObject } from '@stardust-ui/react-component-ref'
+import * as _ from 'lodash'
+import * as React from 'react'
+
+class ReferenceProxy {
+  ref: React.RefObject<HTMLElement>
+
+  constructor(refObject) {
+    this.ref = refObject
+  }
+
+  getBoundingClientRect() {
+    return _.invoke(this.ref.current, 'getBoundingClientRect', {})
+  }
+
+  get clientWidth() {
+    return this.getBoundingClientRect().width
+  }
+
+  get clientHeight() {
+    return this.getBoundingClientRect().height
+  }
+}
+
+/**
+ * Popper.js does not support ref objects from `createRef()` as referenceElement. If we will pass
+ * directly `ref`, `ref.current` will be `null` at the render process. We use memoize to keep the
+ * same reference between renders.
+ *
+ * @see https://popper.js.org/popper-documentation.html#referenceObject
+ */
+const createPopperReferenceProxy = _.memoize(
```

Comment:
how much gain this optimization introduces? Am I right that this helps to omit object being created only, or there is something else I am missing?

Reply:
> Am I right that this helps to omit object being created only

Exactly 👍

Reply:
why we would need it then? I see it as micro optimization that doesn't introduce much value - we might just safely omit that and simplify the logic

Reply:
https://github.com/FezVrasta/react-popper/blob/master/src/Popper.js#L166
To avoid useless calculations in Popper

Reply:
this will create an object that will be stored for the entire app's lifetime in the `memoize's` map, for every Popup content that was once rendered (or asked to be rendered). Lets avoid this problem.